code-name-taran/src/com/spaghettisoft/component/scrabble/Bag.java

```diff
@@ -0,0 +1,49 @@
+package com.spaghettisoft.component.scrabble;
+
+import java.util.ArrayList;
+import java.util.Random;
+
+public class Bag {
+
+	private ArrayList<Letter> letters;
+
+	Bag() {
+		letters = putAllTheLettersInTheBag();
+	}
+
+	public Letter takeRandomLetterFromTheBag() {
+		Random random = new Random();
+		int index = random.nextInt(letters.size());
+		Letter letter = letters.get(index);
+		letters.remove(index);
+		return letter;
+	}
+
+	public void putLetterInTheBag(Letter letter) {
+		letters.add(letter);
+	}
+
+	public boolean isBagEmpty() {
+		if(letters.isEmpty()) {
+			return true;
+		}
+		return false;
+	}
+
+	public int numberOfLettersInTheBag() {

```

Comment:
When naming methods try to use the shortest meaningful name. Method names should also be verbs, so a better name can be "getLettersCount". "InTheBag" is unnecessary, because the method is part of the "Bag" abstraction ^_^